public/app/plugins/datasource/tempo/metricsSummary.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  DataSourceInstanceSettings,
+  FieldDTO,
+  FieldType,
+  LoadingState,
+  MutableDataFrame,
+  sortDataFrame,
+} from '@grafana/data';
+
+export type MetricsSummary = {
+  spanCount: string;
+  errorSpanCount?: string;
+  p50: string;
+  p90: string;
+  p95: string;
+  p99: string;
+  series: Series[];
+};
+
+type Series = {
+  key: string;
+  value: {
+    type: number;
+    n?: number;
+    f?: number;
+    s?: string;
+    b?: string;
+    d?: string;
+    status?: number;
+    kind?: number;
+  };
+};
+
+type MetricsData = {
+  spanCount: string;
+  errorPercentage: string;
+  p50: string;
+  p90: string;
+  p95: string;
+  p99: string;
+  [key: string]: string | number;
+};
+
+export function createTableFrameFromMetricsSummaryQuery(
+  data: MetricsSummary[],
+  targetQuery: string,
+  instanceSettings: DataSourceInstanceSettings,
+  state: LoadingState
+) {
+  let frame;
+
+  if (state === LoadingState.Error) {
+    frame = emptyResponse;
+  } else if (state === LoadingState.Loading) {
+    frame = new MutableDataFrame({
+      name: 'Metrics Summary',
+      refId: 'metrics-summary',
+      fields: [
+        { name: 'state', type: FieldType.string, config: { displayNameFromDS: 'State', custom: { width: 300 } } },
+      ],
+      meta: {
+        preferredVisualisationType: 'table',
+      },
+    });
+    frame.add({
+      state: 'Loading...',
+    });
+  } else if (state === LoadingState.Done && data?.length) {
+    const dynamicMetrics: Record<string, FieldDTO> = {};
+    data.map((res: MetricsSummary) => {
+      const configQuery = getConfigQuery(res.series, targetQuery);
+      res.series.map((series: Series) => {
+        return !dynamicMetrics[series.key]
+          ? (dynamicMetrics[series.key] = {
+              name: `${series.key}`,
+              type: FieldType.string,
+              config: getConfig(series, configQuery, instanceSettings),
+            })
+          : dynamicMetrics[series.key];
+      });
+    });
+
+    frame = new MutableDataFrame({
```

Comment:
As MutableDataFrame is deprecated we should probably use something else to create the Frames. Either manually or through the new helper functions that exists for it.